pkg/postgres/pgutils/retry.go

```diff
@@ -0,0 +1,37 @@
+package pgutils
+
+import (
+	"context"
+	"time"
+
+	"github.com/stackrox/rox/pkg/retry"
+)
+
+const (
+	timeout = 5 * time.Minute
+)
+
+// RetryQuery is used to specify how long to retry to successfully run a query
+// that fails with Transient errors
+func RetryQuery(ctx context.Context, fn func() error) error {
+	if err := ctx.Err(); err != nil {
+		return err
+	}
+	closure := func() error {
+		err := fn()
+		if isTransientError(err) {
+			return retry.MakeRetryable(err)
+		}
+		return err
+	}
+	numTries := int(timeout / 1 * time.Second)
+	err := retry.WithRetry(closure,
```

Comment:
one comment here is that we have a 15s connect timeout so maybe retrying in this way isn't the best